pages/api/hakkasan.ts

```diff
@@ -18,11 +18,12 @@ const handler = async (
     case 'GET': {
       try {
         if (process.env.NODE_ENV !== 'production') {
-          await delay(2000);
-          res.status(200).send({
-            artists,
-            events,
-            venues,
+          await delay(1000, () => {
```

Comment:
Is this callback necessary? If we're `await`ing on some delay, wouldn't it be cleaner simply to make the `res.send` call after the `await` instead of passing it wrapped in an anonymous function?

Reply:
Hmm, this refactor was made so that it was easier to test.

Reply:
I'd prefer a cleaner API over easier unit tests, especially here since requiring a callback introduces further nesting and overhead when reading the code.

If we were to redefine the delay function back to something like this:
```ts
export const delay = (ms: number): Promise<void> =>
  new Promise((resolve) => setTimeout(() => resolve(), ms));
```

We can test delayed code execution using a Promise chain to call a mock function after a delay, advancing the time, and resolving all jobs in the [internal queue `PromiseJobs`](https://javascript.info/microtask-queue) and _then_ testing whether a function has been called:
```ts
  test('calls code after the given ms', async () => {
    const callAfterTimeout = jest.fn();
    actualDelay(2000).then(callAfterTimeout);        // create a promise chain to resolve after 2000ms

    jest.advanceTimersByTime(2000);                  // advance time by 2000ms
    await Promise.resolve();                         // run all pending callbacks in PromiseJobs
    expect(callAfterTimeout).toHaveBeenCalled();     // code should have been called
  });
```

We can also tweak this test to make it a negative test:
```ts
  test('does not call code before the given ms', async () => {
    const callAfterTimeout = jest.fn();
    actualDelay(2000).then(callAfterTimeout);        // create a promise chain to resolve after 2000ms

    jest.advanceTimersByTime(1000);                  // advance time by only 1000ms
    await Promise.resolve();                         // run all pending callbacks in PromiseJobs
    expect(callAfterTimeout).not.toHaveBeenCalled(); // code should not have been called
  });
```